### ISSUE TITLE
dnsdbq high CPU usage while waiting for curl

### ISSUE THREAD

**Issue by djw1149** [dnsdb/dnsdbq#29]
dnsdbq has high CPU usage while waiting for curl.
an strace on debian 9.4 shows:

```...
rt_sigaction(SIGPIPE, {SIG_DFL, [], SA_RESTORER, 0x7f7c515cb1a0}, NULL, 8) = 0
poll([{fd=3, events=POLLIN}], 1, 0)     = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {1890211, 429542592}) = 0
rt_sigaction(SIGPIPE, NULL, {SIG_DFL, [], SA_RESTORER, 0x7f7c515cb1a0}, 8) = 0
rt_sigaction(SIGPIPE, {SIG_IGN, [], SA_RESTORER, 0x7f7c515cb1a0}, NULL, 8) = 0
poll([{fd=3, events=POLLIN|POLLPRI|POLLRDNORM|POLLRDBAND}], 1, 0) = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {1890211, 429709728}) = 0
clock_gettime(CLOCK_MONOTONIC, {1890211, 429749857}) = 0
clock_gettime(CLOCK_MONOTONIC, {1890211, 429853999}) = 0
rt_sigaction(SIGPIPE, {SIG_DFL, [], SA_RESTORER, 0x7f7c515cb1a0}, NULL, 8) = 0
poll([{fd=3, events=POLLIN}], 1, 0)     = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {1890211, 429998273}) = 0
rt_sigaction(SIGPIPE, NULL, {SIG_DFL, [], SA_RESTORER, 0x7f7c515cb1a0}, 8) = 0
rt_sigaction(SIGPIPE, {SIG_IGN, [], SA_RESTORER, 0x7f7c515cb1a0}, NULL, 8) = 0
poll([{fd=3, events=POLLIN|POLLPRI|POLLRDNORM|POLLRDBAND}], 1, 0) = 0 (Timeout)
...
```

**Comment by vixie** [comment]
the code that's doing this is here:

/* io_engine -- let libcurl run until there are few enough outstanding jobs.
 */
static void
io_engine(int jobs) {
        int still;

        if (debuglev > 1)
                fprintf(stderr, "io_engine(%d)\n", jobs);

        /* let libcurl run while there are too many jobs remaining. */
        still = 0;
        while (curl_multi_perform(multi, &still) == CURLM_OK && still > jobs) {
                if (debuglev > 3)
                        fprintf(stderr, "...waiting (still %d)\n", still);
                curl_multi_wait(multi, NULL, 0, 0, NULL);
        }

        /* drain the response code reports. */
        still = 0;
        while (curl_multi_info_read(multi, &still) != NULL) {
                if (debuglev > 3)
                        fprintf(stderr, "...info read (still %d)\n", still);
        }
}

it's my first time using curl_multi. extra eyeballs are clearly needed.

**Comment by vixie** [comment]
fixed in 2e52c37bb78ad2cb9a6057d924288bd1c944cac8
